Allow Specification of Task Role ARN for ECS Slave Templates

ukayani opened the pull request:
Sorry for creating a second pull request for the same feature. It seems like the other PR is not moving.

- I've used the changes by `joh-m`
- Updated the logic to check if task role is specified and only then, add it to the register task def request
- Added a helpfile for the taskRoleArn field

**A few notes**:

I've packaged this plugin and tested it on a Jenkins master running on Amazon ECS. 
There is a gotcha with regards to passing a role ARN to a task:
- The role assumed by the jenkins master (task or ec2-instance) needs to have an `iam:PassRole` policy for any roles which you wish to assign to a slave task

Otherwise you'll get an error similar to the following:

`User: A is not authorized to perform: iam:PassRole on resource: B`
Where A is the arn of the role assumed by the jenkins Master and B is the arn of the role you are trying to specify via taskRoleArn

An example policy statement for the Jenkins Master Task to have permission to assign task roles would look like:

```json
{
            "Effect": "Allow",
            "Action": "iam:PassRole",
            "Resource": "arn:aws:iam::444913385113:role/*"
}
```

- I changed the name of the field from `taskRoleArn` to `taskrole`. For some reason, when  i used the camel case field name, the corresponding help file was not being picked up. I tried including the case in the helpfile name and remove case but nothing seemed to work. Perhaps someone more experienced with the nuances of Jelly help file resolution could shed some light

roehrijn commented:
Hi @ukayani,

sorry, but I had a cronological look on outstanding pull requests today. This resulted in merging #34 first. As your code adds additional value here may I encourage you to rebase your work? 

And about the include of the helpfile. I can't help out here as I'm not very familiar with jelly too. Maybe the dev-Mailinglist is able to help?

Regards, Jan

ukayani commented:
@roehrijn   Thanks for taking a look. I've rebased my changes onto your master. Should be able to merge now.

I've renamed the taskRoleArn field to taskrole due to an issue with resolving the jelly helpfile,  it appears that camel casing in the advanced section does not work as expected. Please let me know if that name change is not acceptable.

Regards,
Umair

stewartbryson commented:
Guys... just trying to decide whether or not to apply this update, but I'm having issues understanding what it does, and the risk mentioned above.

ukayani commented:
Hi @stewartbryson

What is your concern? 

This update allows you to specify a task role ARN in the event that your slave needs permissions to specific AWS resources. It is an optional field so not using it does nothing. You may opt to give your ecs container instance permissions to access any AWS resource or you can choose to give granular permissions to slaves. This option allows you to do the latter
